src/d3d11/d3d11_context_ext.cpp

```diff
@@ -134,4 +139,459 @@ namespace dxvk {
+      return nullptr;
+    return (ID3D11SamplerState *)got->second;
+  }
+  void D3D11DeviceContextExt::add_srv_and_handle(ID3D11ShaderResourceView *srv_imageview, uint32_t handle) {
+    s_SrvHandleToPtr[handle] = srv_imageview;
+  }
+  ID3D11ShaderResourceView *D3D11DeviceContextExt::handle_to_srv(uint32_t handle) {
+    auto got = s_SrvHandleToPtr.find(handle);
+    if (got == s_SrvHandleToPtr.end())
+      return nullptr;
+    return (ID3D11ShaderResourceView *)got->second;
+  }
+
+
+  bool STDMETHODCALLTYPE D3D11DeviceContextExt::GetCudaTextureObject(uint32_t srvDriverHandle, uint32_t samplerDriverHandle, uint32_t* pCudaTextureHandle) {
+    D3D10DeviceLock lock = m_ctx->LockContext();
+
+    ID3D11ShaderResourceView* srv = handle_to_srv(srvDriverHandle);
+    if (!srv) {
+      Logger::warn(str::format("GetCudaTextureObject() failure - srv handle wasn't found: ", srvDriverHandle));
+      return false;
+    }
+
+    ID3D11SamplerState* samplerstate = handle_to_sampler(samplerDriverHandle);
+    if (!samplerstate) {
+      Logger::warn(str::format("GetCudaTextureObject() failure - sampler handle wasn't found: ", samplerDriverHandle));
+      return false;
+    }
+
+    dxvk::Rc<dxvk::DxvkDevice> dxvkdevice = m_ctx->m_device;
+    VkDevice vkDevice = dxvkdevice->handle();
+
+    D3D11SamplerState* pSS = static_cast<D3D11SamplerState*>(samplerstate);
+    Rc<DxvkSampler> pDSS = pSS->GetDXVKSampler();
+    VkSampler vk_sampler = pDSS->handle();
+
+    D3D11ShaderResourceView* pSRV = static_cast<D3D11ShaderResourceView*>(srv);
+    Rc<DxvkImageView> pIV = pSRV->GetImageView();
+    VkImageView vk_imageview = pIV->handle();
+
+    VkImageViewHandleInfoNVX imageViewHandleInfo = {VK_STRUCTURE_TYPE_IMAGE_VIEW_HANDLE_INFO_NVX};
+    imageViewHandleInfo.imageView = vk_imageview;
+    imageViewHandleInfo.sampler = vk_sampler;
+    imageViewHandleInfo.descriptorType = VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER;
+
+    // note: there's no implicit lifetime management here; it's up to the
+    // app to keep the sampler and SRV alive as long as it wants to use this
+    // derived handle.
+
+    *pCudaTextureHandle = dxvkdevice->vkd()->vkGetImageViewHandleNVX(vkDevice, &imageViewHandleInfo);
+
+    if (!*pCudaTextureHandle) {
+      Logger::warn("GetCudaTextureObject() handle==0 - failed");
+      return false;
+    }
+
+    return true;
+  }
+
+
+  class CubinShaderProxy : public RcObject {
+  public:
+    CubinShaderProxy(Rc<dxvk::DxvkDevice> in_dxvkDevice)
+    : rcDxvkDevice(in_dxvkDevice) {
+    };
+    ~CubinShaderProxy() {
+      VkDevice vkDevice = rcDxvkDevice->handle();
+      rcDxvkDevice->vkd()->vkDestroyCuFunctionNVX(vkDevice, Function, nullptr);
+      rcDxvkDevice->vkd()->vkDestroyCuModuleNVX(vkDevice, Module, nullptr);
+    };
+    VkCuModuleNVX Module;
+    VkCuFunctionNVX Function;
+    uint32_t BlockDimX, BlockDimY, BlockDimZ;
+    Rc<DxvkDevice> rcDxvkDevice;
+  };
+
+  class CubinShaderLaunchInfo : public RcObject {
```

Comment:
Does this need to be ref-counted? I know this is a somewhat large struct, but it's probably better to just capture by value using `std::move` in order to avoid vector copies.

Reply:
You're right, refcounting was a leftover from before reorganization.  I've removed that now.  I believe I'm capturing by pointer.